frontend/hooks/useBalanceContext.ts

```diff
@@ -203,11 +205,26 @@ export const useMasterBalances = () => {
 
   /**
    * Unstaked balances across master safes and eoas
+   * Only contains wallets related to the selected agent's home chain id
```

Comment:
the problem was here, the masterSafeBalances are filtered based on `selectedAgentConfig.requiresAgentSafesOn.includes(evmChainId)` which is not correct when for some chains we don't have master safes created or if they are different

I had to introduce a separate `allMasterWalletBalances` not to break things and added a TODO to refactor it all